x-pack/plugins/task_manager/server/saved_objects/migrations.test.ts

```diff
@@ -108,6 +108,45 @@ describe('successful migrations', () => {
       });
     });
   });
+
+  describe('8.2.0', () => {
+    test('resets attempts and status of the failed alerting tasks without schedule interval', () => {
+      const migration820 = getMigrations()['8.2.0'];
+      const taskInstance = getMockData({
+        taskType: 'alerting:123',
+        status: 'failed',
+        schedule: undefined,
+      });
+
+      expect(migration820(taskInstance, migrationContext)).toEqual({
+        ...taskInstance,
+        attributes: {
+          ...taskInstance.attributes,
+          attempts: 0,
+          status: 'idle',
+        },
+      });
+    });
+
+    test('does not update the tasks that are not "failed"', () => {
+      const migration820 = getMigrations()['8.2.0'];
+      const taskInstance = getMockData({
+        taskType: 'alerting:123',
+        status: 'idle',
+        attempts: 3,
+        schedule: undefined,
+      });
+
+      expect(migration820(taskInstance, migrationContext)).toEqual({
```

Comment:
Might be clearer to do `expect(migration820(taskInstance, migrationContext)).toEqual(taskInstance)`?

Reply:
Makes sense :)